core/modes/language_modes.py

```diff
@@ -74,20 +20,162 @@
-    for ext in extensions.split()
-}
-
-language_ids = set(language_extensions.keys())
+extension_lang_map = None
+
+language_ids = None
+language_extensions = None
+
+SETTINGS_FILENAME = "language_modes.csv"
+settings_filepath = compute_csv_path(SETTINGS_FILENAME)
+
+LANGUAGE_HEADERS = ["language", "extensions", "spoken_forms"]
+
+
+def make_sure_settings_file_exists():
+    # Maps language mode names to the extensions that activate them and language spoken forms. Only put things
+    # here which have a supported language mode; that's why there are so many
+    # commented out entries. TODO: make this a csv file?
+    default_csv_contents = [
+        # ['assembly', ('asm', 's'),],
+        # ['bash', ('bashbook', 'sh'),],
+        [
+            "batch",
+            ("bat",),
+        ],
+        [
+            "c",
+            ("c", "h"),
+        ],
+        # ['cmake', ('cmake',),],
+        # ["cplusplus", ("cpp hpp",), ("see plus plus",)],
+        ["csharp", ("cs",), ("see sharp",)],
+        ["css", ("css",), ("c s s",)],
+        # ['elisp', ('el'),],
+        # ['elm', ('elm'),],
+        ["gdb", ("gdb",), ("g d b",)],
+        ["go", ("go",), ("go", "go lang", "go language")],
+        ["java", ("java",)],
+        ["javascript", ("js",)],
+        ["javascriptreact", ("jsx",)],
+        # ["json", ("json",),],
+        [
+            "kotlin",
+            ("kt",),
+        ],
+        [
+            "lua",
+            ("lua",),
+        ],
+        [
+            "markdown",
+            ("md",),
+        ],
+        # ['perl', ('pl',),],
+        [
+            "php",
+            ("php",),
+        ],
+        # ['powershell', ('ps1',),],
+        [
+            "python",
+            ("py",),
+        ],
+        [
+            "protobuf",
+            ("proto",),
+        ],
+        ["r", ("r"), ("are language",)],
+        # ['racket', ('rkt,'),],
+        [
+            "ruby",
+            ("rb",),
+        ],
+        [
+            "rust",
+            ("rs",),
+        ],
+        [
+            "scala",
+            ("scala",),
+        ],
+        [
+            "scss",
+            ("scss",),
+        ],
+        # ['snippets', ('snippets',),],
+        [
+            "sql",
+            ("sql",),
+        ],
+        [
+            "stata",
+            ("do", "ado"),
+        ],
+        [
+            "talon",
+            ("talon",),
+        ],
+        [
+            "talonlist",
+            ("talon-list",),
+        ],
+        [
+            "terraform",
+            ("tf",),
+        ],
+        ["tex", ("tex",), ("tech", "lay tech", "latex")],
+        [
+            "typescript",
+            ("ts",),
+        ],
+        [
+            "typescriptreact",
+            ("tsx",),
+        ],
+        # ['vba', ('vba',),],
+        [
+            "vimscript",
+            ("vim", "vimrc"),
+        ],
+        # htm doesn't actually have a language moded, but we do have snippets.
```

Comment:
Please add "l" and remove "d" after the word "mode"

Reply:
I fixed the comment.